### ISSUE TITLE
Clarification on 527 classes in AudioSet inference pipeline

### ISSUE THREAD

**Issue by Jiaxin-Ye** [XiaoMi/dasheng#11]
Hello,

I am currently working with the AudioSet dataset and am using the inference pipeline for classification. However, I am unsure about the order of the 527 categories in the inference pipeline. Could you please clarify how these categories are indexed or provide a reference to where I can find the specific order?

Thank you in advance for your help!

**Comment by Jiaxin-Ye** [comment]
Could you confirm if this label indices config is used in your work? I am looking forward to your response! 

https://github.com/IBM/audioset-classification/blob/master/audioset_classify/metadata/class_labels_indices.csv

**Comment by RicherMans** [comment]
Hey @Jiaxin-Ye,
Sorry for the late reply.

Yeah that's correct, but note that for audio set there is a fine tuned model. The pre trained dasheng is a pure encoder and has no classification capabilities.

The fine tuned model has an mAP of 49.7 and can be used like this:

```python
from typing import Any, Mapping
import dasheng
import torch

class DashengAudiosetClassifier(torch.nn.Module):

    def __init__(self) -> None:
        super().__init__()
        self.dashengmodel = dasheng.dasheng_base()
        self.classifier = torch.nn.Sequential(torch.nn.LayerNorm(self.dashengmodel.embed_dim), torch.nn.Linear(self.dashengmodel.embed_dim, 527))

    def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True, assign: bool = False):
        self.dashengmodel.load_state_dict(state_dict, strict=False)
        for_classifier_dict = {}
        for k,v in state_dict.items():
            if 'outputlayer' in k:
                for_classifier_dict[k.replace('outputlayer.','')]  = v
        self.classifier.load_state_dict(for_classifier_dict)
        return self

    def forward(self, x):
        x = self.dashengmodel(x).mean(1)
        return self.classifier(x).sigmoid()


mdl = DashengAudiosetClassifier()
check = torch.hub.load_state_dict_from_url('https://zenodo.org/records/13315686/files/dasheng_audioset_mAP497.pt?download=1',map_location='cpu')
mdl.load_state_dict(check)

prediction = mdl(torch.randn(1,16000))
```
